compiler/rustc_middle/src/mir/query.rs

```diff
@@ -360,7 +364,9 @@ pub enum ConstraintCategory<'tcx> {
     ///
     /// We try to get the category that the closure used when reporting this.
     ClosureBounds,
-    CallArgument(Option<(DefId, SubstsRef<'tcx>)>),
+
+    /// Contains the function type if available.
+    CallArgument(Option<Ty<'tcx>>),
```

Comment:
I wonder if we can get the static size back down by changing this to be `CallArgument(Box<Option<Ty<'tcx>>)`, because `Ty<'tcx>` is 40 bytes, and I don't know for sure whether boxing will help or not in that case.

Reply:
We can't afaict, we need `ConstraintCategory` to be `Copy`.